### ISSUE TITLE
Unresolved import `std::panic::AssertUnwindSafe`

### ISSUE THREAD

**Issue by djc** [alexcrichton/curl-rust#126]
In https://bugs.gentoo.org/show_bug.cgi?id=591942, there are reports of some people getting this error while trying to build cargo-0.12.0 or cargo-0.13.0. However, building it seems to work fine for me. Do you have any clues about how the contents of `std::panic` might vary?

``` rust
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:3:24: 3:40 error: unresolved import `std::panic::AssertUnwindSafe`. There is no `AssertUnwindSafe` in `std::panic` [E0432]
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:3 use std::panic::{self, AssertUnwindSafe};
                                                                                                                                                 ^~~~~~~~~~~~~~~~
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:3:24: 3:40 help: run `rustc --explain E0432` to see a detailed explanation
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:17:11: 17:30 error: unresolved name `panic::catch_unwind` [E0425]
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:17     match panic::catch_unwind(AssertUnwindSafe(f)) {
                                                                                                                                     ^~~~~~~~~~~~~~~~~~~
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:17:11: 17:30 help: run `rustc --explain E0425` to see a detailed explanation
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:28:9: 28:29 error: unresolved name `panic::resume_unwind` [E0425]
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:28         panic::resume_unwind(t)
                                                                                                                                   ^~~~~~~~~~~~~~~~~~~~
/var/tmp/paludis/dev-util-cargo-0.12.0/work/cargo_home/registry/src/github.com-88ac128001ac3a9a/curl-0.3.0/src/panic.rs:28:9: 28:29 help: run `rustc --explain E0425` to see a detailed explanation
Build failed, waiting for other jobs to finish...
error: aborting due to 3 previous errors
```

**Comment by alexcrichton** [comment]
Thanks for the report! The `AssertUnwindSafe` type was added in Rust 1.9.0, so if you're compiling a recent version of this crate with an older version of the compiler, it'll cause this error. What version of the compiler is being used in this case?

**Comment by alexcrichton** [comment]
Closing this as I believe it's intended.
